sympy/solvers/ode/single.py

```diff
@@ -651,3 +653,80 @@ def _get_general_solution(self, *, simplify: bool = True):
             ) * exp(-(1 - n)*Integral(P, x)))**(1/(1 - n))
         )
         return [gensol]
+
+
+class Factorable(SingleODESolver):
+
+    hint = "factorable"
+    has_integral = False
+
+    def _matches(self):
+        eq = self.ode_problem.eq
+        f = self.ode_problem.func.func
+        x = self.ode_problem.sym
+        order =self.ode_problem.order
+        df = f(x).diff(x)
+        eqs = []
+        eq = eq.collect(f(x), func = cancel)
+        eq = fraction(factor(eq))[0]
+        roots = factor_list(eq)[1]
+        if len(roots)>1 or roots[0][1]>1:
+            for i in roots:
+                for j in range(1,int(i[1])+1):
+                    if i[0].has(f(x)):
+                        eqs.append(i[0])
```

Comment:
What is the loop over `j` doing here? It looks like it repeatedly adds the same equation.

Reply:
yes it is to add same equation to the list. 
for example:
 
    df = f(x).diff(x)
    eq = df**2 + 2*df + 1
    eqs = [(df+1),(df+1)] 
    
    
 
here the value of `i` will be `((df + 1),2)`. as the `factor_list` gives answers in the form `(base,power)`

Reply:
But why add the same equation twice? We will get the same solutions by adding it once and then we only need to call dsolve recursively once.